internal/copy/copy_dir.go

```diff
@@ -87,22 +85,35 @@ func CopyDir(dst, src string) error {
 			return nil
 		}
 
-		// If the current path is a symlink, recreate the symlink relative to
-		// the dst directory
-		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
-			target, err := os.Readlink(path)
-			if err != nil {
-				return fmt.Errorf("failed to read symlink %q: %w", path, err)
+		errg.Go(func() error {
+
+			// we don't want to try and copy the same file over itself.
+			if eq, err := SameFile(path, dstPath); err != nil {
```

Comment:
Could you elaborate on why you moved this into the errgroup instead of having it before the `mkdir`?

Reply:
I identified SameFile as a bit expensive in the TOFU_CPU_PROFILEing I did.  It makes a bunch of syscalls which are better in the go routine.

The mkdir I left out as I didn't want them contending to make dirs and sub dirs.  It was also a very small amount of the overall runtime.